plumbing/object/tree.go

```diff
@@ -77,7 +77,7 @@ func (t *Tree) File(path string) (*File, error) {
 
 	blob, err := GetBlob(t.s, e.Hash)
 	if err != nil {
-		return nil, err
+		return nil, ErrFileNotFound
```

Comment:
There might be errors other than file not found (such as decoding or network errors). So if `err == plumbing.ErrObjectNotFound then ErrFileNotFound`, but otherwise just return the error.